lnwire/fuzz_test.go

```diff
@@ -822,3 +822,160 @@ func FuzzCustomMessage(f *testing.F) {
 		harness(t, data)
 	})
 }
+
+// prefixWithFailCode adds a failure code prefix to data.
+func prefixWithFailCode(data []byte, code FailCode) []byte {
+	var codeBytes [2]byte
+	binary.BigEndian.PutUint16(codeBytes[:], uint16(code))
+	data = append(codeBytes[:], data...)
+
+	return data
+}
+
+// onionFailureHarness performs the actual fuzz testing of the appropriate onion
+// failure message. This function will check that the passed-in message passes
+// wire length checks, is a valid message once deserialized, and passes a
+// sequence of serialization and deserialization checks.
+func onionFailureHarness(t *testing.T, data []byte, code FailCode) {
+	data = prefixWithFailCode(data, code)
+
+	// Don't waste time fuzzing messages larger than we'll ever accept.
+	if len(data) > MaxSliceLength {
```

Comment:
because these are onion-encoded, I think that the max size is much smaller than 65535 bytes. I would have to calculate what the exact value is, but we can get in more fuzzing runs if this value is smaller

Reply:
Do you mean the 256 byte size prescribed in [BOLT 4](https://github.com/lightning/bolts/blob/master/04-onion-routing.md#requirements-4)?

When we encode, we always produce 256-byte packets or fail if we can't.  But when decoding, we must accept packets of any size >= 256 bytes.

Reply:
Ah no, I was thinking of the hop payloads, nvm